lib/facility.rb

```diff
@@ -1,14 +1,94 @@
+require 'registrant'
+require 'date'
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+# changed to accept single argument as hash named facility_info
+# added .fetch to use for default service value
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
+    @services = facility_info.fetch(:services, [])
+    @registered_vehicles = []
+    @collected_fees = 0
   end
-
-  def add_services(service)
+# add_services changed to add_service 
+  def add_service(service)
     @services << service
   end
+  # administer_written_test method to update license_data
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') 
+    registrant.license_data[:written] = true
+    return true
+    else 
+    return false
+    end
+  end
+# administer_road_test method to update license_data
+  def administer_road_test(registrant)
+    if @services.include?('Road Test')
+      registrant.license_data[:license] = true
+      return true
+    else
+      return false
+    end
+  end
+# renew_drivers_license method to update license_data
+  def renew_drivers_license(registrant)
+    if @services.include?('Renew License')
+      registrant.license_data[:renewed] = true
+      return true
+    else
+      return false
+    end
+  end
+  # register_vehicle method to register vehicles
+  # determines fee collection based on plat_type
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      vehicle.registration_date = Date.today
+  
+      if vehicle.engine == :ev
```

Comment:
Refactor Opportunity (for the future, we know time is limited!!): Consider, when you have a long method like this, how you can use helper methods to break up the logic. For example, you could have a helper method here that you call `process_antique_registration` and it can handle setting the plate type, setting registration date, collecting the right amount of fees for this type of car, etc.

Using helper methods can help a lot when we have complex or nested conditionals, because those get hard to read. If you call helper methods though, you're able to extract some of that logic into a separate location and give it a readable name to indicate to another dev what is happening.